File: hw/pazmino/u2/HW25GUIAndMongo/ContactBookWithMongo/src/main/java/ec/edu/espe/contactbook/controller/ContactController.java
package ec.edu.espe.contactbook.controller;

import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import ec.edu.espe.contactbook.model.Contact;
import ec.edu.espe.contactbook.model.MongoConnection;
import java.util.ArrayList;
import java.util.Calendar;
import org.bson.Document;
import utils.MongoDbManager;

/**
 *
 * @author Brandon Pazmino
 */
public class ContactController {
 private MongoCollection<Document> collection;

    public ContactController() {
       
        MongoDatabase database = new MongoConnection().createConnection().getDatabase("OOP");
        this.collection = database.getCollection("CONTACTSBOOK");
    }

 
    public boolean add(Contact contact) {
        boolean added = false;
     
        if (MongoDbManager.add(contact)) {
            added = true;
        }
        return added;
    }

    public Object find(String findString) {
     
        ArrayList<String> favoriteSport = new ArrayList<>();
        favoriteSport.add("Chess");
        Calendar dateOfBirth = Calendar.getInstance();
        dateOfBirth.set(Calendar.YEAR, 2020);
        dateOfBirth.set(Calendar.MONTH, 1);
        dateOfBirth.set(Calendar.DAY_OF_MONTH, 1);
        Object object = new Contact(3, "Brandon", 'm', "man", "single", dateOfBirth, favoriteSport, "he has a problem");
        return object;
    }
}

File: hw/pazmino/u2/HW25GUIAndMongo/ContactsBookPython/ec/edu/espe/contact_book/controller/contact_controller.py
from datetime import datetime
from utils import MongoDbManager
from model import Contact

class ContactController:
    def __init__(self):
        self.collection = MongoDbManager.connect()  # Usa MongoDbManager para la conexión

    def add(self, contact):
        return MongoDbManager.add(contact)

    def find(self, find_string):
        favorite_sport = ["Chess"]
        date_of_birth = datetime(2020, 2, 1)  # 1 de febrero de 2020
        return Contact(3, "Brandon", 'm', "man", "single", date_of_birth, favorite_sport, "he has a problem")
